apps/web/src/middleware.ts

```diff
@@ -0,0 +1,18 @@
+import { NextResponse, type NextRequest } from 'next/server'
+
+import env from './env'
+import { apiUnauthorized } from './helpers/api/errors'
+
+export function middleware(request: NextRequest) {
+  const { headers } = request
+  const [type, token] = headers.get('Authorization')?.split(' ') ?? []
+  if (type !== 'Bearer' || token !== env.LATITUDE_API_KEY) {
+    return apiUnauthorized()
+  }
+
+  return NextResponse.next()
+}
+
+export const config = {
+  matcher: '/api/v1/(.*)',
+}
```

Comment:
We're not using middleware. Are we?

Reply:
wip

Reply:
I see 2 API endpoints. Can you explain how it works? 
![image](https://github.com/user-attachments/assets/76b2e1ed-e702-47ac-bdc6-1fe3c3f166d3)

Reply:
hmm the one at the bottom looks like old code for when we used client store with regular api, can you remove it? we don't really need it anymore! (sorry i don't have time to work on this atm)

Reply:
nevermind just removed it myself